sympy/integrals/manualintegrate.py

```diff
@@ -887,18 +917,23 @@ def sqrt_linear_rule(integral: IntegralInfo):
         return step
 
 
-def sqrt_quadratic_denom_rule(integral: IntegralInfo):
+def sqrt_quadratic_denom_rule(integral: IntegralInfo, degenerate=True):
     integrand, x = integral
-    numer, denom = integrand.as_numer_denom()
     a = Wild('a', exclude=[x])
     b = Wild('b', exclude=[x])
     c = Wild('c', exclude=[x, 0])
-    match = denom.match(sqrt(a+b*x+c*x**2))
+    numer = Wild('numer')
+    match = integrand.match(numer/sqrt(a+b*x+c*x**2))
     if match:
+        a, b, c, numer = match[a], match[b], match[c], match[numer]
         numer_poly = numer.as_poly(x)
         if numer_poly is None:
             return
-        a, b, c = match[a], match[b], match[c]
+        generate_cond = Ne(c, 0)
+        degenerate_step = None if not degenerate or generate_cond is S.true else \
+            sqrt_linear_rule(IntegralInfo(numer/sqrt(a+b*x), x)) if b.is_zero is not True else \
+            integral_steps(numer/sqrt(a), x)
```

Comment:
Do you find this kind of code easy to understand?

Something like this is much better just written as a plain if/elif:
```python
if not degenerate or generate_cond is S.true:
    degenerate_step = None
elif b.is_zero is not True:
    degenerate_step = sqrt_linear_rule(IntegralInfo(numer/sqrt(a+b*x), x))
else:
    degenerate_step = integral_steps(numer/sqrt(a), x)
```
Then it's easy to see all of the cases clearly. The syntax, spacing and indentation make it easy to visually separate the conditions from the rules that apply for those conditions.

I would only use a ternary if/else if the whole expression is short and fits neatly on one line like:
```python
absx = x if x >= 0 else -x
```
Even then I'm not really sure I'd use it but I can sort of see the appeal of having something short that fits on a line.

Reply:
Honestly I didn't think much. It's indeed not good for non-trivial cases.
And it's not good for testing coverage either.